src/main/java/com/neuvector/Scanner.java

```diff
@@ -329,6 +339,74 @@ private static String maskCredential(String message, String credential){
         return message.replace(credential, "******");
     }
 
+    static String[] getDockerGroupCmdArgs(String scanReportPath, Boolean includeUserId) {
+        String[] cmdGroupArgs = {"", ""};
+        if(!includeUserId){
+            return cmdGroupArgs;
+        }
+
+        // No user arg if file exists, and it is owned by root
+        Boolean ownedByRoot = ownedByRoot(scanReportPath);
+        if (ownedByRoot != null && ownedByRoot) {
+            return cmdGroupArgs;
+        }
+
+        String UID = executeCommand("id -u");
+        if ((UID != null && !UID.isEmpty())) {
+            cmdGroupArgs[0] = "-u";
+            cmdGroupArgs[1] = UID;
+        }
+        return cmdGroupArgs;
+    }
+
+    private static Boolean ownedByRoot(String scanReportPath)  {
+        File scanResultFileJson = new File(scanReportPath);
+        try {
+            return scanResultsFileExist(scanReportPath) && getUserPrincipal(scanReportPath, scanResultFileJson).getName().equals("root");
```

Comment:
It is better to change this to 

```
"root".equals(getUserPrincipal(scanReportPath, scanResultFileJson).getName()); 
```

to avoid potential NPE.

Reply:
ca5e6d18a0e63e8efa5d2b641b99737be51f6674